src/rules/extensions.js

```diff
@@ -46,9 +46,10 @@ module.exports = {
       typeof configuration === 'object' ? configuration : context.options[1]
     )
 
-    function isUseOfExtensionRequired(extension) {
+    function isUseOfExtensionRequired(extension, isModule) {
       if (!has(modifiers, extension)) { modifiers[extension] = defaultConfig }
-      return modifiers[extension] === 'always'
+      const modifier = modifiers[extension]
+      return (!isModule && modifier === 'ignorePackages') || modifier === 'always'
```

Comment:
Good catch!  I'm not well versed in all of the possible import permutations, but here's a regex I just made up.  Can you tell me if I'm missing something?

```
^(([\w]((?!\/).)*)|(@[^\/]+\/?[^\/]+))$
```

This matches the following:

```
foo
foo2
foo.js
@foo
@foo2
@foo.js
@foo/bar
@foo2/bar
@foo2.js/bar
@foo2.js/bar.js
```

And will not match the following:

```
foo/bar
foo2/bar
foo/bar2
foo/bar.js
foo/bar/hello
@foo2/bar/hello
@foo2/bar/hello.js
```